revapi-java/src/test/java/org/revapi/java/JavaArchiveAnalyzerTest.java

```diff
@@ -85,19 +87,28 @@ public void testSimple() throws Exception {
     @Test
     public void testWithSupplementary() throws Exception {
         ArchiveAndCompilationPath compRes = createCompiledJar("a.jar", "v1/supplementary/a/A.java",
-                "v1/supplementary/b/B.java", "v1/supplementary/a/C.java");
+                "v1/supplementary/b/B.java", "v1/supplementary/b/D.java");
 
         JavaArchive api = ShrinkWrap.create(JavaArchive.class, "api.jar")
-                .addAsResource(compRes.compilationPath.resolve("A.class").toFile(), "A.class");
+                .addAsResource(compRes.compilationPath.resolve(A_PACKAGE_PATH + "A.class").toFile(),
+                        A_PACKAGE_PATH + "A.class")
+                .addAsResource(compRes.compilationPath.resolve(A_PACKAGE_PATH + "A$PrivateEnum.class").toFile(),
+                        A_PACKAGE_PATH + "A$PrivateEnum.class");
         JavaArchive sup = ShrinkWrap.create(JavaArchive.class, "sup.jar")
-                .addAsResource(compRes.compilationPath.resolve("B.class").toFile(), "B.class")
-                .addAsResource(compRes.compilationPath.resolve("B$T$1.class").toFile(), "B$T$1.class")
-                .addAsResource(compRes.compilationPath.resolve("B$T$1$TT$1.class").toFile(), "B$T$1$TT$1.class")
-                .addAsResource(compRes.compilationPath.resolve("B$T$2.class").toFile(), "B$T$2.class")
-                .addAsResource(compRes.compilationPath.resolve("C.class").toFile(), "C.class")
-                .addAsResource(compRes.compilationPath.resolve("B$UsedByIgnoredClass.class").toFile(),
-                        "B$UsedByIgnoredClass.class")
-                .addAsResource(compRes.compilationPath.resolve("A$PrivateEnum.class").toFile(), "A$PrivateEnum.class");
+                .addAsResource(compRes.compilationPath.resolve(SUP_PACKAGE_PATH + "B.class").toFile(),
+                        SUP_PACKAGE_PATH + "B.class")
+                .addAsResource(compRes.compilationPath.resolve(SUP_PACKAGE_PATH + "B$T$1.class").toFile(),
+                        SUP_PACKAGE_PATH + "B$T$1.class")
+                .addAsResource(compRes.compilationPath.resolve(SUP_PACKAGE_PATH + "B$T$1$TT$1.class").toFile(),
+                        SUP_PACKAGE_PATH + "B$T$1$TT$1.class")
+                .addAsResource(compRes.compilationPath.resolve(SUP_PACKAGE_PATH + "B$T$2.class").toFile(),
+                        SUP_PACKAGE_PATH + "B$T$2.class")
+                // Change to D because this is used in the api in SupplementaryJarsTest (split package fixes)
```

Comment:
I think this was here exactly because Java8- allowed split packages (and I think Java8+ still supports it when only classpath is used).

Reply:
In this case we are considering java 17, where this is enforced.

Reply:
Yes, that's right. But even in the original coverage is kept in java 8.

Reply:
We can readd that runs only for java 8 to cover that case.

Reply:
@metlos readded a test that runs only in java 8 for verifying split package allowed case with supplementary jars.